src/DGtal/kernel/UnorderedSetByBlock.h

```diff
@@ -30,177 +30,48 @@
-    /// coordinates and its bit within this block.
-    ///
-    /// @param e any lattice point
-    ///
-    /// @return a pair grouping its block coordinates and its bit
-    /// within this block.
-    static
-    std::pair< Element, DGtal::Dimension >
-    split( const Element& e )
-    {
-      Element se = e;
-      se[ 0 ]   &= 0xffffffe0;
-      return std::make_pair( se, (DGtal::Dimension) e[ 0 ] & 0x0000001f );
-    }
-
-    /// Rejoins a splitted element (see \ref split).
-    ///
-    /// @param e the block coordinate of the element.
-    /// @param x the bit coordinate of the element.
-    ///
-    /// @return the corresponding rejoined element.
-    static
-    Element
-    join( const Element& e, DGtal::Dimension x )
-    {
-      Element ge = e;
-      ge[ 0 ] |= x;
-      return ge;
-    }
-
-    /// Rejoins a splitted element (see \ref split).
-    ///
-    /// @param p a pair grouping the block coordinate and the bit
-    /// coordinate of the element.
-    ///
-    /// @return the corresponding rejoined element.
-    static
-    Element
-    join( const std::pair< Element, DGtal::Dimension >& p )
-    {
-      Element ge = p.first;
-      ge[ 0 ] |= p.second; 
-      return ge;
-    }
-  }; // struct Splitter< PointVector< dim, DGtal::int32_t, TContainer > >
-
-  /// Splits an integral-array element into an integral-array element
-  /// and a 32 bit integer. The expected behaviour is the following one for an element e
-  /// of dimension 3:
   /// \code
-  /// Splitter< Point > split;
+  /// Splitter< Point,uint64_t > split; // for 64 bits word
   /// auto v = split( Point( 117, 43, 52 ) );
-  /// v.first  == 117 & 0x1f
-  /// v.second == Point( 117 & 0xffff...ffe0, 43, 52 );
+  /// v.first  == 117 & 63
+  /// v.second == Point( 117 - v.first, 43, 52 );
   /// \endcode
   ///
-  /// @tparam dim the dimension of the integral array.
-  /// @tparam TContainer the container for PointVector.
-  ///
-  /// @note In subclass, we use mask operations instead of mult/div 32. The result
-  /// 4 x times faster !
+  /// @tparam TElement the type of array-like element.
   ///
-  /// @note Specialization for 64 bits integer types
+  /// @tparam TWord the type for holding blocks (uint64_t or uint32_t generally),
+  /// a model of concepts::CIntegralNumber and concepts::CUnsignedNumber.
   ///
   /// @see UnorderedSetByBlock
-  template < DGtal::Dimension dim, typename TContainer >
-  struct Splitter< PointVector< dim, DGtal::int64_t, TContainer > > {
-    typedef PointVector< dim, DGtal::int64_t, TContainer > Element;
-    typedef Splitter< Element >  Self;
-    typedef uint32_t             Word;
+  template < typename TElement, typename TWord = uint32_t >
+  struct Splitter {
+    typedef Splitter< TElement,TWord >   Self;
+    typedef TElement                     Element;
+    typedef TWord                        Word;
+    typedef typename Element::Coordinate Coordinate;
 
+    BOOST_CONCEPT_ASSERT(( concepts::CIntegralNumber< Word > ));
```

Comment:
```suggestion
    BOOST_CONCEPT_ASSERT(( concepts::CBoundedNumber< Word > ));
```
To remove `BigInteger` from the accepted word types.

Reply:
Tou are right !